### PR TITLE
task2 wasm completed

### PR DESCRIPTION
**Pull request by yyhuang95**

### CODE REVIEW

File: wasm.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+function LoadWasm(cfg){
+    const wasmBuffer = fs.readFileSync('./wasm.wasm');
+    if (cfg != undefined) {
+        return WebAssembly.instantiate(wasmBuffer, cfg)
+    } else {
+        return WebAssembly.instantiate(wasmBuffer)
+    }
+}
+
+describe('test wasm', function() {
```
Comment (KKiiim) [inline wasm.test.js:14]: test did not work in CI action

File: .github/workflows/c-cpp.yml [hunk 2]
```diff
@@ -5,32 +5,23 @@ on:
...
-        name: install qemu
-        run: 
-          sudo apt install qemu-user
-        
-      - name: cross-compile gtest for arm64
-        run: |
-          git clone https://github.com/google/googletest.git && 
-          mkdir gtest-build && 
-          cd gtest-build && cmake ../googletest/ -DCMAKE_INSTALL_PREFIX=/usr/aarch64-linux-gnu -DCMAKE_CXX_COMPILER=/usr/bin/aarch64-linux-gnu-g++ && 
-          make && sudo make install
```
Comment (KKiiim) [inline .github/workflows/c-cpp.yml:29]: remain the old test and append the new, organize src and test files in your repo

**Review by KKiiim — CHANGES_REQUESTED** [review]

File: .github/workflows/wasm.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    branches: [ "yiyang" ]
+jobs:
+  test-wasm:
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: install build essentials
+        run: 
+          sudo apt update && 
+          sudo apt install nodejs
```
Comment (XMadrid) [inline .github/workflows/wasm.yml:16]: it's better to use actions/setup-node

File: .github/workflows/wasm.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      
+      -
+        uses: actions/checkout@v3
+      - 
+        name: install mocha
+        run: |
+          cd wasm &&
+          npm install -g mocha
```
Comment (XMadrid) [inline .github/workflows/wasm.yml:24]: use `npm ci` to install dependencies.

File: .github/workflows/wasm.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      -
+        name: run test
+        run: |
+          npx run mocha wasm.test.js
```
Comment (XMadrid) [inline .github/workflows/wasm.yml:28]: `cd wasm && npm run test`

File: wasm/wasm.test.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                    if (arr[i] < m) {
+                        return false
+                    } else {
+                        m = arr[i]
+                    }
+                }
+                return true
+            }
+            myQuickSort(0, raw_values.length - 1)
+            assert.equal(true, assertQS(mem, 0, raw_values.length))
```
Comment (KKiiim) [inline wasm/wasm.test.js:61]: use mocha test

**Review by KKiiim — CHANGES_REQUESTED** [review]

**Review by KKiiim — APPROVED** [review]

**Pull request merged**